osu.Game.Tests/Visual/Gameplay/TestSceneOverlayActivation.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Game.Configuration;
+using osu.Game.Overlays;
+using osu.Game.Rulesets;
+
+namespace osu.Game.Tests.Visual.Gameplay
+{
+    public class TestSceneOverlayActivation : OsuPlayerTestScene
```

Comment:
would also like a test case for allowing the overlays to be opened by disabling the setting here for completeness

Reply:
In fact, I had a test to ensure that overlays can be opened when the setting is disabled, but I ended up not committing it as for some reason the setting update doesn't propagate to the loaded test player (whereas this works just fine in game), making the test always fail.

Reply:
That is quite unsettling. I'd like to know why exactly it fails in the test.

Reply:
For reference the test was:
```csharp
        [Test]
        public void TestGameplayOverlayActivationSettingDisabled()
        {
            AddStep("enable overlay activation during gameplay", () => config.Set(OsuSetting.GameplayDisableOverlayActivation, false));
            AddAssert("activation mode is user triggered", () => testPlayer.OverlayActivationMode == OverlayActivation.UserTriggered);
        }
```
and would fail on the assertion of the overlay activation mode (always returning `OverlayActivationMode.Disabled` whether it should be `OverlayActivationMode.UserTriggered`).
I tried replacing with an `AddUntil` step but I keep getting the same results.

It looks like the setting update doesn't get propagated to the test player.

You can easily replicate by copy pasting into the test cases file.

Reply:
the test doesn't work because it's using the wrong instance of `OsuConfigManager`. `PlayerTestScene` creates and caches its own:

https://github.com/ppy/osu/blob/c09868780e1d03703a18f37e512777fba3afe89c/osu.Game/Tests/Visual/PlayerTestScene.cs#L25-L32

using that one instead of the locally-`[Resolved]` config manager fixes the failure.

on that note I'd also like to say that I find the mentality of discarding the test if it doesn't work correctly quite dangerous. we shouldn't become accustomed to accepting that things just don't work sometimes for unknown reasons. if I hadn't prodded continuously reviewing this, I probably wouldn't have found out that the tests were half-broken due to using the wrong config instance, especially seeing that they *read* sane.

(also not sure how to feel about having to debug this myself, but, alas.)

Reply:
I apologize for having left the debugging of this to you. I feel really bad for doing it, especially when the cause is that I didn't paid attention enough to the existing code.